### ISSUE TITLE
关于r2r和rxr的数据收集

### ISSUE THREAD

**Issue by RandyP-P** [MIV-XJTU/JanusVLN#28]
你好，感谢你开源这个优秀的项目。
我想问一下项目中提供的收集好的R2R和RxR数据是通过habitat的path_follower收集的吗？
我发现RxR的episode文件序号存在比较明显的断裂，最大序号好像是六万多但实际上只有19705个episode文件。请问收集这个是否使用了特定的过滤规则？
我注意到uni-navid等项目收集数据时会做一些动作合并（比如两个转15度动作合并为转30度）还会平衡各种动作的比例，请问目前提供的轨迹数据有没有经过类似的操作？

**Comment by missTL** [comment]
For the R2R and RxR data, we collected it through the training process of VLN-CE.
We only used data with English instructions, filtering out data in other languages.
We have not implemented action merging yet, but it is a straightforward task that only requires merging consecutive identical actions.

**Comment by RandyP-P** [comment]
Thank you so much for your previous reply!
I have a few more questions,
I am currently fine-tuning the Qwen2.5-VL-3B-Instruct model using the R2R and RxR datasets on an 8x A100 (80GB) server. To evaluate the training efficiency, I ran a 40-step profiling test and would love to get your thoughts on the estimated training time.
Profiling Details
For my setup, I use a per-device batch size of 8 (global batch size = 64). I used torch.cuda.synchronize() to measure the exact time spent in each forward/backward component. Here is the breakdown:
分段                            平均(ms)       占比     最小(ms)     最大(ms)    采样数
----------------------------------------------------------------------
vit_encode                    2838.0    34.0%     2284.6     3660.7     17
history_compress               263.5     3.2%       12.3     1282.8     17
image_scatter                    3.8     0.0%        3.1        5.7     17
rope_build                       0.5     0.0%        0.3        1.9     17
llm_forward                    629.8     7.5%      577.1      806.2     17
backward_only                 4410.0    52.8%
data+framework+other           213.0     2.5%
----------------------------------------------------------------------
平均 step 总耗时                   8358.6 ms

图片数(输入vit的图片数):     min=99, max=158, avg=121.9
序列长度(进入LLM的token数):   min=2097, max=2535, avg=2346.3

With synchronization turned off, a single step takes about 8 seconds.there's about 2.4M samples in r2r and rxr datasets, in terms of this calculation method it takes about 90 hours (or ~720 GPU hours), which feels quite long.
My Questions
1.Could you share roughly how many GPU hours it took you to train on the complete R2R and RxR datasets?
2.Based on my profiling breakdown and the training config below, do you have any recommendations or best practices to optimize the training speed?
Thank you so much for your help!

My training configuration is as follows:
torchrun --nproc_per_node=$NPROC_PER_NODE \
            --master_addr=$MASTER_ADDR \
            --master_port=$MASTER_PORT \
            qwen-vl-finetune/qwenvl/train/train_qwen.py \
            --model_name_or_path $MODEL_PATH \
            --tune_mm_llm True \
            --tune_mm_vision False \
            --tune_mm_mlp True \
            --dataset_use $DATASETS \
            --output_dir $OUTPUT_DIR \
            --cache_dir $CACHE_DIR \
            --bf16 \
            --per_device_train_batch_size 8 \
            --gradient_accumulation_steps 1 \
            --learning_rate 2e-5 \
            --optim adamw_torch \
            --model_max_length 8192 \
            --data_flatten False \
            --max_pixels $((576*28*28)) \
            --min_pixels $((16*28*28)) \
            --base_interval 2 \
            --video_max_frames 8 \
            --video_min_frames 4 \
            --num_train_epochs 1 \
            --warmup_ratio 0.03 \
            --lr_scheduler_type "cosine" \
            --weight_decay 0.01 \
            --logging_steps 10 \
            --save_steps 1000 \
            --save_total_limit 1 \
            --deepspeed "qwen-vl-finetune/scripts/zero2.json" \
            --gradient_checkpointing \
            --dataloader_num_workers 8 \
            --seed 42 \
            --logging_dir "${OUTPUT_DIR}/runs" \
            --report_to "tensorboard" \
            --max_steps 40 \
            2>&1 | tee "${OUTPUT_DIR}/train.log"
